### PR TITLE
Fix logic for getting maintainer ids for benchmark workflow

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
This PR fixes the logic to fetch maintainer id list from `CODEOWNERS` file in `benchmark-pull-request` workflow. 

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- ~[ ] Functionality includes testing.~
- ~[ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.~
- ~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by jed326 — APPROVED** [review]

**Comment by rishabh6788** [comment]
@jed326 your approval should be sufficient since you are one of the maintainers, right?

File: .github/workflows/benchmark-pull-request.yml
Comment (peternied) [inline .github/workflows/benchmark-pull-request.yml:110]: Looks like we are trying to solve this problem in multiple ways, I'd prefer this API based method:
https://github.com/opensearch-project/OpenSearch/blob/6e27e5ad8e37944605bef61e14f1972f440e4724/.github/workflows/maintainer-approval.yml#L11-L32

**Review by peternied — CHANGES_REQUESTED** [review]

File: .github/workflows/benchmark-pull-request.yml
Comment (jed326) [reply .github/workflows/benchmark-pull-request.yml:110]: Thanks @peternied, I think this is the better approach.
I double checked https://github.com/opensearch-project/OpenSearch/pull/15163 and it seems like we aren't adding all the maintainers to https://github.com/opensearch-project/OpenSearch/blob/6e27e5ad8e37944605bef61e14f1972f440e4724/.github/CODEOWNERS#L37 either

**Comment by jed326** [comment]
> @jed326 your approval should be sufficient since you are one of the maintainers, right?

I think PRs need to be approved again if any changes are pushed after approvals

File: .github/workflows/benchmark-pull-request.yml
Comment (rishabh6788) [reply .github/workflows/benchmark-pull-request.yml:110]: This looks way better, will update it.

Comment (rishabh6788) [reply .github/workflows/benchmark-pull-request.yml:110]: @peternied Updated as per recommendation.

**Review by peternied — APPROVED** [review]

**Comment by peternied** [comment]
@rishabh6788 Looks like you can only assign up to 10 people on the issues being created, see #15458.  Consider creating an issue if you'd like to fix that.

- Also I'm not interested in fielding these requests, so I've created https://github.com/opensearch-project/OpenSearch/pull/15468

**Pull request merged**
